pkg/schemaconv/smd.go

```diff
@@ -122,6 +122,56 @@ func (c *convert) makeRef(model proto.Schema) schema.TypeRef {
 	return tr
 }
 
+func makeUnion(extensions map[string]interface{}) (*schema.Union, error) {
+	union := schema.Union{
+		Fields: map[string]string{},
+	}
+
+	if idiscriminator, ok := extensions["x-kubernetes-oneof-discriminator"]; ok {
+		discriminator, ok := idiscriminator.(string)
+		if !ok {
+			return nil, fmt.Errorf(`"x-kubernetes-oneof-discriminator" is of invalid type: %T`, idiscriminator)
+		}
+		union.Discriminator = &discriminator
+	}
+
+	if ifields, ok := extensions["x-kubernetes-oneof-fields"]; ok {
+		fields, ok := ifields.(map[interface{}]interface{})
+		if !ok {
+			return nil, fmt.Errorf(`"x-kubernetes-oneof-fields" is of invalid type: %T`, ifields)
+		}
+		for ifield, value := range fields {
+			field, ok := ifield.(string)
+			if !ok {
+				return nil, fmt.Errorf(`"x-kubernetes-oneof-fields": field is of invalid type: %T`, ifield)
```

Comment:
Definitely need to print the values to make this and the following error actionable.